[EDU-3627] Bot Protection reference (#538)
aziontech/docs: src/content/docs/en/pages/main-menu/reference/marketplace/bot-protection-reference.mdx

    - Defining custom rule management to act on individual bots based on previous content extraction activity from requests.

- **Increase visibility**
    - Measuring the amount and characteristics of the bot traffic trying to access your website, APIs, and applications.
    - Exporting your logs, integrating your SIEM, or using the observability tools provided by Azion to monitor the malicious activity.
    - Combining other integrations to enhance Bot Protection eficiency, through use of fingerprint, captchas, JavaScript injection, or SDKs, to create robust rules.

- **Reduce financial risk**
    - Protecting your website and applications against credential abuse, card balance verification, and other forms of online fraud.

---

## Implementation

| Scope | Description |
|---|---|
| [How to install Azion Bot Protection](/en/documentation/products/guides/bot-protection/) | Guide step-by-step to install and configure the Bot Protection integration. |
| [Azion Integrations](/en/documentation/products/marketplace/integrations/) |	Reference to deepen the basics of Azion Integrations, how they work, and how you can start using them. |

---

## How Azion Bot Protection works

> Azion **Bot Protection** is *serverless* integration available at Azion **Marketplace**.

Reviewer: ```suggestion
Azion **Bot Protection** is a *serverless* integration available at Azion **Marketplace**.
```

request > Edge Firewall > Executa Function > Entrega score > Human or good Bot > request permitida > Logs
<br/>
                                                            se for um Bad bot > action definida > logs

////

1. When a request reaches a domain using Bot Protection, Edge Firewall executes the function.
2. After the analysis, the function defines the score for that request, based on a predefined group of rules.
3. If the score is equals or higher than the predetermined treshold, the predefined action is executed.

Bot Protection is able to execute **5 different actions** whenever the request's score is greater or equals than the defined threshold:

- `allow`: allows the continuation of the request.
- `deny`: delivers a standard *Status Code 403* response.
- `drop`: terminates the request without a response to the user.
- `redirect`: allows the request to be redirected to a new URL/Location when the security threshold is reached. 
- `custom_html`: allows customized HTML content to be delivered to the user in case of a threshold violation.

All this actions can be configured for web and mobile applications, as well as APIs, offering protection in different environments.

> The function also can POST the requested data to any compatible system with HTTP Receiver Protocol, allowing export the logs, as well as integration with SIEM and other Azion integrations and products. 

Reviewer: ```suggestion
The function can also POST the requested data to any compatible system with HTTP Receiver Protocol, allowing export the logs, as well as integration with SIEM and other Azion integrations and products. 
```

:::tip

> You can also benefit from **Secure** products, including Web Application Firewall, DDoS protection unmmetered, Network Layer Protection, and Intelligent DNS, to increase your protection, adding aditional security layers to your infrastructure.

Reviewer: ```suggestion
You can also benefit from **Secure** products, including Web Application Firewall, DDoS Protection Unmetered, Network Layer Protection, and Intelligent DNS, to increase your protection, adding extra security layers to your infrastructure.
```

:::

### Use cases

**Bot Protection** was deveoped by Azion to attent to uses cases envolving common practices of malicious bots and traffic.

- Reputation Intelligence
- Bot attacking
- Account takeover
- Credential stuffing
- Vulnerability scanning
- Brute force attacks
- Web scraping

<Button href="/en/documentation/products/guides/bot-protection/" text="go to how to install Bot Protection guide" variant="secondary"> 

</Button>
---

## Features

> Azion **Bot Protection** is composed by different features that you can take advantage. 

Reviewer: ```suggestion
Azion **Bot Protection** is composed by different features that you can take advantage of.
```

### Reputation Intelligence

By using *Reputation Intelligence*, Bot Protection stablishes an additional security perimeter, cataloging the inbound and outbound traffic, based on **Network Lists**, mantained and constantly updatated by Azion. Through these lists, Bot Protection is able to identify the profile of each request trying to reach your site.

Network Lists used by **Bot Protection** includes criterias such as:

- Tor Exit Nodes
- Reputation
- Proxies
- Malware
- Fraud

<Button href="/en/documentation/products/edge-firewall/network-layer-protection/network-lists/" text="go to network list reference" variant="secondary"> 

</Button>

### Bot classification

Based on the scores and Reputation Intelligence, the integration is able to classificate different types of bots and traffic. 

- Legitimate traffic

- Good bots
    - Social network bot
    - Monitoring bot
    - Aggregator bot
    - Enterprise bot
    - Search engine bot

- Bad Bots
    - Bad bot signatures
    - Malicious intent detected
    - Scripted bots
    - Malicious browser behavior
    - Reputation Intelligence

### Device identification

> Bot Protection leverages advanced techniques to identify and distinguish between legitimate devices and potentially malicious bots attempting to access your digital assets. The integration is capable of generate a user ID for each device.

Reviewer: ```suggestion
Bot Protection leverages advanced techniques to identify and distinguish between legitimate devices and potentially malicious bots attempting to access your digital assets. The integration is capable of generating a user ID for each device.
```

> To further enhance its protective capabilities, it allows for the incorporation of additional security layers through other integrations, such as **SDKs**, **JavaScript injection**, and **fingerprint** to collect more granular data.

Reviewer: ```suggestion
To further enhance its protective capabilities, it allows for the incorporation of additional security layers through other integrations, such as **SDKs**, **JavaScript injection**, and **Fingerprint** to collect more granular data.
```

---

## Additional resources

### SDK

Azion **Bot Protection** can work altogether with Software Development Kits (SDKs), allowing you to customize and tailor security protocols to meet the specific needs of your mobile applications. With SDKs and Bot Prottection, you can implement fine-grained controls, address application-specific vulnerabilities, and adapt to evolving threats more effectively.

You can use SDKs to track mobile devices and identify behaviors (touching the screen, for example) and device data (model, manufacturer, operational system, etc.) to use as insights for Bot Protection detect and mitigate malicious threats. 

### JavaScript injection

When the JavaScript file is inserted in your edge application, it collects data on the actions made by the device used in a request. It's available for use with web browsers. With the JavaScript injection, more data will be collected, such as manufacturer and hardware used, to execute request rules.

This data can be used to create more robust rules and behaviors on the Bot Protection args in order to detect and mitigate threats more effectively.

### Rate Limiting

> Rate limiting integration establishes thresholds for the number of requests a user or system can make within a specified timeframe, effectively mitigating the impact of brute force attacks or excessive bot activities.  By working jointly with rate limiting, the bot protection measures gain an additional layer of defense against automated threats. 

Reviewer: ```suggestion
Rate limiting integration establishes thresholds for the number of requests a user or system can make within a specified timeframe, effectively mitigating the impact of brute force attacks or excessive bot activities. By working jointly with rate limiting, the bot protection measures gain an additional layer of defense against automated threats.
```

- `deny`: delivers a standard *Status Code 403* response.
- `drop`: terminates the request without a response to the user.
- `redirect`: allows the request to be redirected to a new URL/Location when the security threshold is reached. 
- `custom_html`: allows customized HTML content to be delivered to the user in case of a threshold violation.

All this actions can be configured for web and mobile applications, as well as APIs, offering protection in different environments.

The function can also POST the requested data to any compatible system with [HTTP Receiver Protocol](#export-logs), allowing integration with SIEMs and other Azion integrations and products. 

:::tip
You can also benefit from **Secure** products, including [Web Application Firewall](/en/documentation/products/edge-firewall/web-application-firewall/), [DDoS Protection](/en/documentation/products/edge-firewall/ddos-protection/), [Network Layer Protection](/en/documentation/products/edge-firewall/network-layer-protection/), and [Intelligent DNS](/en/documentation/products/intelligent-dns/), to increase your protection, adding extra security layers to your infrastructure.
:::

### Use cases

> **Bot Protection** was deveoped by Azion to attent to uses cases envolving common practices of malicious bots and traffic.

Author: ```suggestion
**Bot Protection** was developed by Azion to attend to uses cases involving common practices of malicious bots and traffic.
```

---
title: Azion Bot Protection
description: Protect your applications against automated attacks such as bad bot signatures or scripted bots.
meta_tags: bot, protection, cybersecurity, edge computing
namespace: docs_bot_protection_reference
permalink: /documentation/products/marketplace/bot-protection/
---

import Button from '~/components/Button.astro';

> Azion **Bot Protection** uses advanced algorithms and *Reputation Intelligence* to analyze the behavior of incoming data. This enables the detection of suspicious traffic and bad bots, facilitating the implementation of preventive measures against malicious activities such as credential stuffing, vulnerability scanning, and site scraping.

Author: ```suggestion
Azion **Bot Protection** uses advanced algorithms, machine learning, and *Reputation Intelligence* to analyze the behavior of incoming data. This enables the detection of suspicious traffic and bad bots, facilitating the implementation of preventive measures against malicious activities such as credential stuffing, vulnerability scanning, and site scraping.
```

> Azion **Bot Protection** uses advanced algorithms, machine learning, and *Reputation Intelligence* to analyze the behavior of incoming data. This enables the detection of suspicious traffic and bad bots, facilitating the implementation of preventive measures against malicious activities such as credential stuffing, vulnerability scanning, and site scraping.

Author: ```suggestion
Azion **Bot Protection** uses advanced intelligent algorithms and *Reputation Intelligence* to analyze the behavior of incoming data. This enables the detection of suspicious traffic and bad bots, facilitating the implementation of preventive measures against malicious activities such as credential stuffing, vulnerability scanning, and site scraping.
```

Azion **Bot Protection** uses advanced intelligent algorithms and *Reputation Intelligence* to analyze the behavior of incoming data. This enables the detection of suspicious traffic and bad bots, facilitating the implementation of preventive measures against malicious activities such as credential stuffing, vulnerability scanning, and site scraping.

This integration assigns a score to every request based on rules, behaviors, and Reputation Intelligence. If the score exceeds a predetermined threshold, the integration executes the predetermined action to deny, drop, or redirect the request. 

By using **Bot Protection**, you can:

- **Enhance user experience**
    - Reducing the impact of bots on the entire infrastructure.
    - Providing bot protection by IP reputation.
    - Defining custom rule management to act on individual bots based on previous content extraction activity from requests.

- **Increase visibility**
    - Measuring the amount and characteristics of the bot traffic trying to access your website, APIs, and applications.
    - Exporting your logs, integrating your SIEM or using the observability tools provided by Azion to monitor the malicious activity.

>     - Combining other integrations to enhance Bot Protection eficiency, through use of fingerprint, captchas, JavaScript injection, or SDKs, to create robust rules.

Reviewer 2: ```suggestion
    - Combining other integrations to enhance Bot Protection efficiency, through the use of fingerprint, captchas, JavaScript injection, or SDKs, to create robust rules.
```

- **Reduce financial risk**
    - Protecting your website and applications against credential abuse, card balance verification, and other forms of online fraud.

---

## Implementation

| Scope | Description |
|---|---|

> | [How to install Azion Bot Protection](/en/documentation/products/guides/bot-protection/) | Guide step-by-step to install and configure the Bot Protection integration. |

Reviewer 3: ```suggestion
| [How to install Azion Bot Protection](/en/documentation/products/guides/bot-protection/) | Step-by-step guide to install and configure the Bot Protection integration. |
```

| [Azion Integrations](/en/documentation/products/marketplace/integrations/) |	Reference to deepen the basics of Azion Integrations, how they work, and how you can start using them. |

---

## How Azion Bot Protection works

Azion **Bot Protection** is a *serverless* integration available at Azion **Marketplace**.

This integration is based on an edge function that runs on [Edge Firewall](/en/documentation/products/edge-firewall/). This type of function takes advantage of the features and modules of **Edge Firewall** to protect your network, applications, and resources from security threats, including bad bot signatures or scripted bots. 

/// GRÁFICO ///s

> ////

Reviewer: rever

1. A request reaches a domain using Bot Protection.
2. Edge Firewall receives the request.
3. The function is executed and starts all the analytics processes, including:

> - Retrieving the request data, including device, browser , and network data, fingerprint, among others.

Author: ```suggestion
- Retrieving the requested data, including device, browser, and network data, fingerprint, among others.
```

- Identifying and classifying the request according to advanced intelligent algorithms and Reputation Intelligence.
- Defining the behavior according to the rules engine criteria.
4. Bot Protection assigns a score to the request.

> 5. If the score is equals or higher than the predetermined threshold, the predefined preventive action is executed.

Reviewer 2: ```suggestion
5. If the score is equal to or higher than the predetermined threshold, the predefined preventive action is executed.
```

:::tip

> You can receive the logs trough an HTTPS endpoint, in JSON format. Read more about [export logs](#export-logs).

Reviewer 2: ```suggestion
You can receive the logs through an HTTPS endpoint, in JSON format. Read more about [export logs](#export-logs).
```

:::

Bot Protection is able to execute **5 different actions** whenever the request's score is greater or equals than the defined threshold:

- `allow`: allows the continuation of the request.
- `deny`: delivers a standard *Status Code 403* response.
- `drop`: terminates the request without a response to the user.

> - `redirect`: allows the request to be redirected to a new URL/Location when the security threshold is reached. 

Reviewer 3: ```suggestion
- `redirect`: allows the request to be redirected to a new URL/location when the security threshold is reached. 
```

- `custom_html`: allows customized HTML content to be delivered to the user in case of a threshold violation.

> All this actions can be configured for web and mobile applications, as well as APIs, offering protection in different environments.

Reviewer 2: ```suggestion
All these actions can be configured for web and mobile applications, as well as APIs, offering protection in different environments.
```

The function can also POST the requested data to any compatible system with [HTTP Receiver Protocol](#export-logs), allowing integration with SIEMs and other Azion integrations and products. 

:::tip
You can also benefit from **Secure** products, including [Web Application Firewall](/en/documentation/products/edge-firewall/web-application-firewall/), [DDoS Protection](/en/documentation/products/edge-firewall/ddos-protection/), [Network Layer Protection](/en/documentation/products/edge-firewall/network-layer-protection/), and [Intelligent DNS](/en/documentation/products/intelligent-dns/), to increase your protection, adding extra security layers to your infrastructure.
:::

### Use cases

> **Bot Protection** was developed by Azion to attend to uses cases involving common practices of malicious bots and traffic.

Reviewer 3: ```suggestion
**Bot Protection** was developed by Azion to attend to use cases involving common practices of malicious bots and traffic.
```

- Reputation Intelligence
- Bot attacking
- Account takeover
- Credential stuffing
- Vulnerability scanning
- Brute force attacks
- Web scraping

<Button href="/en/documentation/products/guides/bot-protection/" text="go to how to install Bot Protection guide" variant="secondary"> 

</Button>
---

## Features

Azion **Bot Protection** is composed by different features that you can take advantage of.

### Reputation Intelligence

> By using *Reputation Intelligence*, Bot Protection stablishes an additional security perimeter, cataloging the inbound and outbound traffic, based on **Network Lists**, mantained and constantly updatated by Azion. Through these lists, Bot Protection is able to identify the profile of each request trying to reach your site.

Reviewer 2: ```suggestion
By using *Reputation Intelligence*, Bot Protection establishes an additional security perimeter, cataloging the inbound and outbound traffic, based on **Network Lists**, maintained and constantly updated by Azion. Through these lists, Bot Protection is able to identify the profile of each request trying to reach your site.
```

> Network Lists used by **Bot Protection** includes criterias such as:

Reviewer 2: ```suggestion
Network Lists used by **Bot Protection** include criterias such as:
```

- Tor Exit Nodes
- Reputation
- Proxies
- Malware
- Fraud

<Button href="/en/documentation/products/edge-firewall/network-layer-protection/network-lists/" text="go to network lists reference" variant="secondary"> 

</Button>

### Bot classification

> Based on the scores and Reputation Intelligence, the integration is able to classificate different types of bots and traffic. 

Reviewer 2: ```suggestion
Based on the scores and Reputation Intelligence, the integration is able to classify different types of bots and traffic. 
```

- Legitimate traffic

- Good bots
    - Social network bot
    - Monitoring bot
    - Aggregator bot
    - Enterprise bot
    - Search engine bot

- Bad Bots
    - Bad bot signatures
    - Malicious intent detected
    - Scripted bots
    - Malicious browser behavior
    - Reputation Intelligence

### Device identification

Bot Protection leverages advanced techniques to identify and distinguish between legitimate devices and potentially malicious bots attempting to access your digital assets. The integration is capable of generating a user ID for each device.

To further enhance its protective capabilities, it allows for the incorporation of additional security layers through other integrations and resources, such as [SDKs](#sdks), [JavaScript injection](#javascript-injection), and [Fingerprint](#fingerprint) to collect more granular data.

### Redirect

> One of the actions Bot Protection is able to execute is `redirect`. It allows the request to be redirected to a new URL/location, specified in the JSON Args, when the security threshold is reached.

Reviewer 3: ```suggestion
One of the actions Bot Protection is able to execute is `redirect`. It allows the request to be redirected to a new URL/location, specified in the JSON args, when the security threshold is reached.
```

### Custom HTML

The integration allows customized HTML content to be delivered to the user in case of a threshold violation, thanks to the `custom_html` action. You can create a custom message to exhibit to users in case of threshold violation.

### Delay response

It allows for the maintenance of an open connection and delaying responses in cases where bots need to function autonomously for extended periods. This eliminates the need for continuous responses, streamlining operations and ensuring consistent performance and reliability in critical tasks.

### Modes

> Azion **Bot Protection** allows you to define the environment in which the function is expected to run, being API or an web application the possible modes. The default mode is `web`. If any value other than the string "api" (case-sensitive in lowercase) is provided, the "web" mode will be used as default configuration.

Reviewer 2: ```suggestion
Azion **Bot Protection** allows you to define the environment in which the function is expected to run, being API or a web application the possible modes. The default mode is `web`. If any value other than the string "api" (case-sensitive in lowercase) is provided, the "web" mode will be used as the default configuration.
```

> Azion **Bot Protection** allows you to define the environment in which the function is expected to run, being API or an web application the possible modes. The default mode is `web`. If any value other than the string "api" (case-sensitive in lowercase) is provided, the "web" mode will be used as default configuration.

Reviewer 3: ```suggestion
Azion **Bot Protection** allows you to define the environment in which the function is expected to run, being API or an web application the possible modes. The default mode is `web`. If any value other than the string `api` (case-sensitive in lowercase) is provided, the `web` mode will be used as default configuration.
```

By enabling the `api` mode, no "Set-Cookie" will be performed, and any rules related to the use of cookies in Bot Protection will be ignored.

### Export logs

By exporting JSON with logs to an HTTPS edpoint, you can maintain a detailed record of interactions, empowering you with the information needed to make informed decisions and continually enhance the security posture of your systems.

When a log is generated, you'll receive a JSON object similar to this: 

```json
{
  "geoip_region": "SP",
  "action": "deny",
  "asn": "396982",
  "request_id": "9581e2b51b5a082b12fed308f4eae564",
  "host": "ojgayey.map.azionedge.com",
  "host": "xxxxxxx.map.azionedge.com",
  "classified": "bad bot",
  "geoip_country": "BR",
  "http_user_agent": "curl/7.87.0",
  "bot category": [
    "Bad Bot Signatures"
  ],
  "request_uri": "/a",
  "remote_addr": "34.95.175.91",
  "score": 8
}
```

> This feature provides comprehensive monitoring, post-event analysis, and near real-time tracking of potential threats or suspicious activities. You can use your prefered endpoint, including an SIEM and Azion **Data Streaming** endpoints to receive the information.

Reviewer 2: ```suggestion
This feature provides comprehensive monitoring, post-event analysis, and near real-time tracking of potential threats or suspicious activities. You can use your preferred endpoint, including an SIEM and Azion **Data Streaming** endpoints to receive the information.
```

> With Data Streaming endpoints, you're able to feed your stream processing, SIEM, and big data platforms with the event logs from your applications on Azion in real time. This product allows you to collect your logs from and connect them to the endpoint of your data analysis platforms. You can also decide which data you want to use on your analysis by choosing among the available variables.

Reviewer: ```suggestion
With Data Streaming connectors, you're able to feed your stream processing, SIEM, and big data platforms with the event logs from your applications on Azion in real time. This product allows you to collect your logs from and connect them to the endpoint of your data analysis platforms. You can also decide which data you want to use on your analysis by choosing among the available variables.
```

<Button href="/en/documentation/products/data-streaming/" text="go to data streaming reference" variant="secondary"> 

</Button>

---

## Additional resources

### SDKs

> Azion **Bot Protection** can work altogether with Software Development Kits (SDKs), for both Android and iOS systems, allowing you to customize and tailor security protocols to meet the specific needs of your mobile applications. With SDKs and Bot Protection, you can implement fine-grained controls, address application-specific vulnerabilities, and adapt to evolving threats more effectively.

Reviewer 3: ```suggestion
Azion **Bot Protection** can work together with Software Development Kits (SDKs), for both Android and iOS systems, allowing you to customize and tailor security protocols to meet the specific needs of your mobile applications. With SDKs and Bot Protection, you can implement fine-grained controls, address application-specific vulnerabilities, and adapt to evolving threats more effectively.
```

You can use SDKs to track mobile devices and identify behaviors (touching the screen, for example) and device data (model, manufacturer, operational system, etc.) to use as insights for Bot Protection detect and mitigate malicious threats. 

### JavaScript injection

> When the JavaScript file is inserted in your edge application, it collects data on the actions made by the device used in a request. It's available for use with web browsers. With the JavaScript injection, more data will be collected, such as manufacturer and hardware used, to execute request rules.

Reviewer 3: ```suggestion
When the JavaScript file is inserted in your edge application, it collects data on the actions made by the device used in a request. It's available for use with web browsers. With JavaScript injection, more data will be collected, such as manufacturer and hardware used, to execute request rules.
```

This data can be used to create more robust rules and behaviors on the Bot Protection args in order to detect and mitigate threats more effectively.

### Rate Limiting

Rate limiting integration establishes thresholds for the number of requests a user or system can make within a specified timeframe, effectively mitigating the impact of brute force attacks or excessive bot activities. By working jointly with rate limiting, the bot protection measures gain an additional layer of defense against automated threats.

### Fingerprint

> A set of information (IP, User-Agent) creates a hash for devices accessing your edge applications. The information is gathered by tracing the device's session and provides a more accurate detailing of the request's device, increasing the precision of Bot Protection logs.

Reviewer 3: ```suggestion
A set of information (IP, `User-Agent` header) creates a hash for devices accessing your edge applications. The information is gathered by tracing the device's session and provides a more accurate detailing of the request's device, increasing the precision of Bot Protection logs.
```

### Captcha 

> By using the `redirect` action, the defined URL/location can contain a Captcha integration to add an additional security layer. It helps you to increase security and malicious traffic detection, challenging all the request previously violating any treshold to guarantee is legitimate.

Reviewer 2: ```suggestion
By using the `redirect` action, the defined URL/location can contain a Captcha integration to add an additional security layer. It helps you to increase security and malicious traffic detection, challenging all the request previously violating any threshold to guarantee is legitimate.
```
